plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,390 @@
+        type: bool
+        default: False
+    return_fingerprints:
+        description:
+            - Allows for the return of fingerprint(s) for newly created or deleted keys(s)
+        type: bool
+        default: False
+'''
+
+EXAMPLES = '''
+- name: Generate the default GPG keypair (Ed25519)
+  community.crypto.gpg_keypair:
+
+- name: Generate the default GPG keypair with a passphrase
+  community.crypto.gpg_keypair:
+    passphrase: super_secret_password
+
+- name: Generate a RSA GPG keypair with the default RSA size (2048 bits)
+  community.crypto.gpg_keypair:
+    key_type: RSA
+
+- name: Generate a RSA GPG keypair with custom size (4096 bits)
+  community.crypto.gpg_keypair:
+    key_type: RSA
+    key_length: 4096
+
+- name: Generate an ECC GPG keypair 
+  community.crypto.gpg_keypair:
+    key_type: EDDSA
+    key_curve: cv25519
+
+- name: Generate a GPG keypair and with a subkey:
+  community.crypto.gpg_keypair:
+    subkey_type: ECDH
+    subkey_curve: cv25519
+
+- name: Generate a GPG keypair with custom user-id:
+  community.crypto.gpg_keypair:
+    name: Your Name
+    comment: Interesting comment.
+    email: example@email.com
+
+- name: Generate a GPG keypair and return fingerprint of new key
+  community.crypto.gpg_keypair:
+    return_fingerprints: true
+  register: gpg_keys
+
+- name: Delete GPG keypair(s) matching a specified user-id:
+  community.crypto.gpg_keypair:
+    state: abscent
+    name: Your Name
+    comment: Interesting comment.
+    email: example@email.com
+
+- name: Delete GPG keypair(s) matching a specified fingerprint:
+  community.crypto.gpg_keypair:
+    state: abscent
+    fingerprints:
+      - ABC123...
+
+'''
+
+RETURN = '''
+size:
+    description: Size (in bits) of the SSH private key.
+    returned: changed or success
+    type: int
+    sample: 4096
+fingerprints:
+    description: Fingerprint(s) of newly created or deleted key(s)
+    return: changed and `return_fingerprints`==True
+    type: list[str]
+    sample: [ ABC123... ]
+'''
+
+from typing import Dict, Union
+
+import itertools
+import re
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.crypto.plugins.module_utils.gnupg.cli import GPGError
+from ansible_collections.community.crypto.plugins.plugin_utils.gnupg import GPGError
+
+def validate_params(params: Dict[str, Union[str, int]]) -> None:
+
+    if params['override'] and params['present'] and not (params['fingerprint'] or params['name'] or params['comment'] or params['email']):
+            raise GPGError, 'To override existing keys, please provide any combination of the `fingerprint`, `name`, `comment`, and `email` parameters.'
+    keys = ['key']
+    if params['subkey_type']:
+        keys.append('subkey')
+    for key in keys:
+        if params[f'{key}_type'] == 'EDDSA':
+            if not params[f'{key}_usage']: params[f'{key}_usage'] = ['sign', 'auth']
```

Comment:
`f` strings don't work with Python 2 either. You need to use `%` or `str.format()`.

Reply:
I'll need do some refactor as I used quite a few f strings haha. Refactoring will be commited later today.